src/Business Foundation/App/NoSeriesCopilot/src/Copilot/NoSeriesProposal.Page.al

```diff
@@ -0,0 +1,197 @@
+                    trigger OnAction()
+                    begin
+                        InputText := CreateNoSeriesForModuleWithPatternLbl;
+                        CurrPage.Update();
+                    end;
+                }
+                action(NewNumberSeriesForCompany)
+                {
+                    ApplicationArea = All;
+                    Caption = 'Create numbers series for the new company';
+                    trigger OnAction()
+                    begin
+                        InputText := CreateNoSeriesForCompanyLbl;
+                        CurrPage.Update();
+                    end;
+                }
+            }
+            group(ModifyExistingNoSeries)
+            {
+                Caption = 'Modify existing';
+
+                action(ChangeNumberTo)
+                {
+                    Caption = 'Change the [sales order] number to [SO-10001]';
+                    trigger OnAction()
+                    begin
+                        InputText := ChangeNumberLbl;
+                        CurrPage.Update();
+                    end;
+                }
+            }
+        }
+
+        area(SystemActions)
+        {
+            systemaction(Generate)
+            {
+                Caption = 'Generate';
+                Tooltip = 'Generate no. series';
+                trigger OnAction()
+                begin
+                    GenerateNoSeries();
+                end;
+            }
+            systemaction(Regenerate)
+            {
+                Caption = 'Regenerate';
+                Tooltip = 'Regenerate no. series';
+                trigger OnAction()
+                begin
+                    GenerateNoSeries();
+                end;
+            }
+            systemaction(Cancel)
+            {
+                ToolTip = 'Discards all suggestions and dismisses the dialog';
+            }
+            systemaction(Ok)
+            {
+                Caption = 'Keep it';
+                ToolTip = 'Accepts the current suggestion and dismisses the dialog';
+            }
+        }
+    }
+
+    var
+        InputText: Text;
+        ResponseText: Text;
+        PageCaptionLbl: text;
+        IsResponseTextVisible: Boolean;
+        IsProposalDetailsVisible: Boolean;
+        CreateNoSeriesForLbl: Label 'Create number series for ';
+        CreateNoSeriesForModuleWithPatternLbl: Label 'Create number series for [specify here] module in the format ';
+        CreateNoSeriesForCompanyLbl: Label 'Create numbers series for the new company';
+        ChangeNumberLbl: Label 'Change the [specify here] number to ';
+
+    trigger OnAfterGetCurrRecord()
+    begin
+        PageCaptionLbl := Rec.GetInputText();
+    end;
+
+    trigger OnQueryClosePage(CloseAction: Action): Boolean
+    var
+    begin
+        if CloseAction = CloseAction::OK then
+            ApplyProposedNoSeries();
+    end;
+
+    local procedure GenerateNoSeries()
```

Comment:
these methods should be moved to a codeunit... and we should have a oneliner here at most..